### PR TITLE
Add Talisman of the Caveman

### PR DESCRIPTION
**Pull request by svr333**
## Description
I added a new talisman. This talisman has a 40% chance to give you haste for 40 seconds.
Heavily based on the Talisman of the Traveller. This idea also came from the #approved section in Discord.

## Changes
- Add new `SlimefunItem` with name `TALISMAN_HASTIER` / `hastier_talisman` unlocked with 20 levels
- Add entry in `messages_en` and `researches_en`
- Add EventListener for the BreakBlockEvent in `TalismanListener.java`

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
<br>
These are not viable in this case, or so I believe  

- [x] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [x] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Comment by Seggan** [comment]
Suggestion: Call it the Talisman of the Miner

**Comment by svr333** [comment]
> 
> 
> Suggestion: Call it the Talisman of the Miner

Talisman of the miner already exists and has a 20% chance to double ores

**Comment by Seggan** [comment]
Hmm "Hastier" doesn't sound right.

**Comment by svr333** [comment]
Well I am open to name suggestions, hastier sounded funny in my head so I used it (its also correct english so)

**Comment by Seggan** [comment]
It's an adjective not a noun so the way you used it is incorrect.

**Comment by svr333** [comment]
Any other name suggestion then?

**Comment by Seggan** [comment]
"Talisman of Urgency" perhaps

**Comment by svr333** [comment]
Mmm could do, Ill leave it open for a while in case anyone comes up with something extraordinaire

**Comment by LinoxGH** [comment]
"Talisman of Haste" or "Talisman of Hastiness" could work too.

**Comment by svr333** [comment]
What about Talisman of the Hurried?

This is to keep in trend with the rest, talisman of the <person type>

**Comment by Lehinx** [comment]
Tailsman of Expert Miner

**Comment by svr333** [comment]
Talisman of the Cavemen?

File: src/main/resources/languages/messages_en.yml [hunk 1]
```diff
@@ -151,6 +151,7 @@ messages:
     knight: '&a&oYour Talisman gave you 5 Seconds of Regeneration'
     whirlwind: '&a&oYour Talisman reflected the Projectile'
     wizard: '&a&oYour Talisman has given you a better Fortune Level but maybe also lowered some other Enchantment Levels'
+    hastier: '&a&oYour Talisman gave you haste'
```
Comment (LinoxGH) [inline src/main/resources/languages/messages_en.yml:154]: These messages should have a '.' in the end of the sentence.

Comment (svr333) [reply src/main/resources/languages/messages_en.yml:154]: @LinoxGH I followed the other talismans here, none of them had a `.` at the end.
I could update all of them though, if you'd like (if I do this, in this PR or make a new one for that?)

Comment (LinoxGH) [reply src/main/resources/languages/messages_en.yml:154]: > @LinoxGH I followed the other talismans here, none of them had a `.` at the end.
> I could update all of them though, if you'd like (if I do this, in this PR or make a new one for that?)

Oh, well it's up to you then. The whole language file and item lores need a revise anyway.

**Review by LinoxGH — APPROVED** [review]

File: src/main/resources/languages/messages_en.yml (hunk 1 above)
Comment (svr333) [reply src/main/resources/languages/messages_en.yml:154]: For this PR Ill keep it like the rest, Ill see if I open a new one to add punctuation to the rest

**Comment by svr333** [comment]
I decided to go with caveman for consistency in the naming of talismans

**Review by LinoxGH — APPROVED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java [hunk 2]
```diff
@@ -546,6 +546,7 @@ private SlimefunItems() {}
...
     public static final SlimefunItemStack TALISMAN_WHIRLWIND = new SlimefunItemStack("WHIRLWIND_TALISMAN", Material.EMERALD, "&aTalisman of the Whirlwind", "", "&fHaving this Talisman", "&fin your Inventory will reflect", "&f60% of any projectiles fired at you.", "&e&oOnly a thrown Trident can pierce", "&e&othrough this layer of protection");
     public static final SlimefunItemStack TALISMAN_WIZARD = new SlimefunItemStack("WIZARD_TALISMAN", Material.EMERALD, "&aTalisman of the Wizard", "", "&fWhile you have this Talisman", "&fin your Inventory it allows you to", "&fobtain Fortune Level 4/5 however", "&fit also has a chance to lower the", "&fLevel of some Enchantments on your Item");
+    public static final SlimefunItemStack TALISMAN_CAVEMAN = new SlimefunItemStack("CAVEMAN_TALISMAN", Material.EMERALD, "&aTalisman of the Caveman", "&fWhile you have this Talisman", "&fin your inventory it gives", "&fyou a 40% Chance for a decent", "&fHaste buff when you mine a block");
```
Comment (poma123) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java:549]: "Chance" should be lowercase.
```suggestion
    public static final SlimefunItemStack TALISMAN_CAVEMAN = new SlimefunItemStack("CAVEMAN_TALISMAN", Material.EMERALD, "&aTalisman of the Caveman", "&fWhile you have this Talisman", "&fin your inventory it gives", "&fyou a 40% chance for a decent", "&fHaste buff when you mine a block");
```

**Review by poma123 — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java (hunk 2 above)
Comment (LinoxGH) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java:549]: But in other talismans it isn't see Talisman of Knight

Comment (svr333) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java:549]: Yeah I followed the other talisman and in there it is also capitalized

Comment (poma123) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java:549]: See Miner Talisman then...

Comment (poma123) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java:549]: Either way, it should be lowercase.

**Review by poma123 — APPROVED** [review]

**Review by LinoxGH — DISMISSED** [review]

**Comment by svr333** [comment]
How come the url validator fails all of a sudden?

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java [hunk 3]
```diff
@@ -284,6 +285,11 @@ public void onBlockDropItems(BlockDropItemEvent e) {
         }
     }
 
+    @EventHandler
+    public void onBlockBreak(BlockBreakEvent e) {
+        Talisman.checkFor(e, SlimefunItems.TALISMAN_CAVEMAN);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java:290]: For redundancy but foremost also for performance reasons, there should probably be a check first whether the Player already has a `Haste` effect running.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java [hunk 4]
```diff
@@ -546,6 +546,7 @@ private SlimefunItems() {}
...
     public static final SlimefunItemStack TALISMAN_WHIRLWIND = new SlimefunItemStack("WHIRLWIND_TALISMAN", Material.EMERALD, "&aTalisman of the Whirlwind", "", "&fHaving this Talisman", "&fin your Inventory will reflect", "&f60% of any projectiles fired at you.", "&e&oOnly a thrown Trident can pierce", "&e&othrough this layer of protection");
     public static final SlimefunItemStack TALISMAN_WIZARD = new SlimefunItemStack("WIZARD_TALISMAN", Material.EMERALD, "&aTalisman of the Wizard", "", "&fWhile you have this Talisman", "&fin your Inventory it allows you to", "&fobtain Fortune Level 4/5 however", "&fit also has a chance to lower the", "&fLevel of some Enchantments on your Item");
+    public static final SlimefunItemStack TALISMAN_CAVEMAN = new SlimefunItemStack("CAVEMAN_TALISMAN", Material.EMERALD, "&aTalisman of the Caveman", "&fWhile you have this Talisman", "&fin your inventory it gives", "&fyou a 40% chance for a decent", "&fHaste buff when you mine a block");
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java:549]: For performance/balancing reasons: Maybe only when an ore is mined and not for every block?
But then increase the chance to perhaps 45%?

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Comment by TheBusyBiscuit** [comment]
> How come the url validator fails all of a sudden?

Rebase your fork off of master, github changed some docs links 🙄

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java (hunk 3 above)
Comment (svr333) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java:290]: This check is done inside of the Talisman class, more specifically in the pass() method that gets called from within the checkFor() method

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java (hunk 4 above)
Comment (svr333) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java:549]: Sure

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java [hunk 5]
```diff
@@ -546,6 +546,7 @@ private SlimefunItems() {}
...
     public static final SlimefunItemStack TALISMAN_WHIRLWIND = new SlimefunItemStack("WHIRLWIND_TALISMAN", Material.EMERALD, "&aTalisman of the Whirlwind", "", "&fHaving this Talisman", "&fin your Inventory will reflect", "&f60% of any projectiles fired at you.", "&e&oOnly a thrown Trident can pierce", "&e&othrough this layer of protection");
     public static final SlimefunItemStack TALISMAN_WIZARD = new SlimefunItemStack("WIZARD_TALISMAN", Material.EMERALD, "&aTalisman of the Wizard", "", "&fWhile you have this Talisman", "&fin your Inventory it allows you to", "&fobtain Fortune Level 4/5 however", "&fit also has a chance to lower the", "&fLevel of some Enchantments on your Item");
+    public static final SlimefunItemStack TALISMAN_CAVEMAN = new SlimefunItemStack("CAVEMAN_TALISMAN", Material.EMERALD, "&aTalisman of the Caveman", "&fWhile you have this Talisman", "&fin your inventory it gives", "&fyou a 40% chance for a decent", "&fHaste buff when you mine any ore");
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java:549]: You forgot to update the lore for the new chance

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java [hunk 6]
```diff
@@ -546,6 +546,7 @@ private SlimefunItems() {}
...
     public static final SlimefunItemStack TALISMAN_WHIRLWIND = new SlimefunItemStack("WHIRLWIND_TALISMAN", Material.EMERALD, "&aTalisman of the Whirlwind", "", "&fHaving this Talisman", "&fin your Inventory will reflect", "&f60% of any projectiles fired at you.", "&e&oOnly a thrown Trident can pierce", "&e&othrough this layer of protection");
     public static final SlimefunItemStack TALISMAN_WIZARD = new SlimefunItemStack("WIZARD_TALISMAN", Material.EMERALD, "&aTalisman of the Wizard", "", "&fWhile you have this Talisman", "&fin your Inventory it allows you to", "&fobtain Fortune Level 4/5 however", "&fit also has a chance to lower the", "&fLevel of some Enchantments on your Item");
+    public static final SlimefunItemStack TALISMAN_CAVEMAN = new SlimefunItemStack("CAVEMAN_TALISMAN", Material.EMERALD, "&aTalisman of the Caveman", "&fWhile you have this Talisman", "&fin your inventory it gives", "&fyou a 50% chance for a decent", "&fHaste buff when you mine any ore");
```
Comment (poma123) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/SlimefunItems.java:549]: One last change I would suggest, to add an empty line between these.
```suggestion
    public static final SlimefunItemStack TALISMAN_CAVEMAN = new SlimefunItemStack("CAVEMAN_TALISMAN", Material.EMERALD, "&aTalisman of the Caveman", "", "&fWhile you have this Talisman", "&fin your inventory it gives", "&fyou a 50% chance for a decent", "&fHaste buff when you mine any ore");
```

**Review by poma123 — CHANGES_REQUESTED** [review]

**Review by poma123 — APPROVED** [review]
LGTM, thanks for your contribution!

**Review by LinoxGH — APPROVED** [review]

**Review by TheBusyBiscuit — APPROVED** [review]

**Comment by TheBusyBiscuit** [comment]
Reminder for future @TheBusyBiscuit to include a new tag for this in PR #2298

**Pull request merged**